### DISCUSSION TITLE
5번째 미팅 때 논의하고 싶은 내용

### DISCUSSION THREAD

**Discussion by Joseph-Cha** [Joseph-Cha/JustFight_Client#16] [Q&A] [answered]
### 1. Pull Request의 독립성
하나의 요구사항에 대해 하나의 Pull Request를 남기는 방식으로 프로젝트를 진행하려 했으나 https://github.com/Joseph-Cha/JustFight_Client/pull/14#issuecomment-781389841 에 남긴 것 처럼 커밋들이 다 한번에 올라가는 사태가 발생합니다 ㅠ 
따라서 브랜치를 각각의 요구사항 별로 생성을 해서 Pull Request를 남기는 것이 좋을지 고민해보면 좋을 것 같습니다. (물론 한 번에 모든 요구상항 별 브랜치를 만든다는게 아니라 현재 작업 중인 요구사항을 브랜치로 생성하는 방법이 어떨지 의견을 나눠보면 좋을 것 같습니다.)

### 2. 닭이 먼저인지 계란이 먼저인지(Behavioral Design과 Class Design)
https://github.com/Joseph-Cha/JustFight_Client/pull/13#discussion_r579815063 여기에 남겨주신 코멘트를 곰곰이 생각해봤는데 결국 behavioral Design을 토대로 Class Design을 하나씩 쌓아나가는 느낌으로 작업이 진행되기를 원하시는 것 같습니다. 제가 생각한 이 방향과 종필님이 원하시는 방향이 맞는지 확인해보는 시간이 있으면 좋을 것 같습니다.

### 3. #13 #14 리뷰

**Comment by jongfeel** [comment, marked as answer]
1. 맞게 한 것 같은데요?? 

이미 commit이 몇 개 진행중인 branch(Design)에서 새롭게 branch(UnitFunction2)를 생성 하면 commit log까지 생성해서 가져가기 때문에 main branch로의 switch를 잘 해서 구분해서 작업하시면 됩니다.

2. behavioural design 기준으로 진행

이 방법이 조금 더 객체 들을 생각해 볼 수 있는 쉬운 방법입니다. 
Collaboration diagram인데

- 객체
- 메시지 전달 (메시지 받는 대상은 메소드를 가짐)
- 실행에 대한 설명

이걸 해보시면 됩니다.

꼭 collaboration diagram으로 안그려도 됩니다.

3. TMI

객체 들의 생성 삭제가 빈번하다면?? (총알이 많이 나가는 게임의 경우)
Active/Deactive 를 해주는 관리 기법을 사용해 보면 좋습니다.
이걸 managing 할 수 있는 객체가 필요.

**Reply by jongfeel** [reply]
4. Encapsulation

Interface 노출?
부모 클래스 타입 노출?
Event 노출?
